hyperspy/signal.py

```diff
@@ -1924,8 +1924,18 @@ def _get_undefined_axes_list(self):
     def __call__(self, axes_manager=None):
         if axes_manager is None:
             axes_manager = self.axes_manager
-        return np.atleast_1d(
-            self.data.__getitem__(axes_manager._getitem_tuple))
+        if getattr(self._plot, 'pointer', None) is not None:
+            indices = self._plot.pointer.indices
+            size = self._plot.pointer.get_size_in_indices()
+            array_slices = [slice(None, None, None)] * \
+                self.axes_manager.navigation_dimension
+            for i, axis in enumerate(self.axes_manager.navigation_axes[::-1]):
```

Comment:
You should not iterate over all navigation axes if the pointer is only defined for the two. I suggest getting the axes from the pointer itself (and store them there if they are not currently).

Additionally, no need to create a `[slice(None)]*nav_dim` list which is then completely replaced. You could, for example, use the `list(axes_manager._getitem_tuple)` and replace just the indices the pointer changes.